static/js/admin.js

```diff
@@ -549,6 +569,41 @@ function _setup_page() {
         });
     });
 
+    $(".admin_user_table, .admin_bot_table").on("click", ".open-user-form", function (e) {
+        var email = $(e.currentTarget).data("email");
+        var user_row = $("tr[id='user_" + email + "']");
+        var user_name = user_row.find(".user_name");
+        var form_row = $("tr[id='user_form_" + email + "'");
+        var reset_button = form_row.find(".reset_edit_user");
+        var submit_button = form_row.find(".submit_name_changes");
+        var full_name = form_row.find("input[name='full_name']");
+
+        // Show user form.
+        user_row.hide();
+        form_row.show();
+
+        reset_button.on("click", function (e) {
+            form_row.hide();
+            user_row.show();
+        });
+
+        submit_button.on("click", function (e) {
+            e.preventDefault();
+            e.stopPropagation();
+
+            var url = "/json/users/" + email;
+            var data = {
+                full_name: JSON.stringify(full_name.val())
+            };
+
+            channel.patch({
+                url: url,
+                data: data,
+                error: failed_changing_name
```

Comment:
I would put `report_success_message` as the success handler here, since only here do we know whether it was this browser tab that actually requested the change.